magento/magento2
# #26314: fixed logic for updating map price for selected swatch

### Description
Fixed logic for updating map price for selected swatch option on product list

### Fixed Issues
1. #26314 

### Manual testing scenarios
1. Enable MAP (Stores -> Configuration -> Sales -> Sales -> Minimum Advertised Price -> Enable MAP).
1. Enable 'Show Swatches in Product List', enabled by default (Stores -> Configuration -> Catalog -> Catalog -> Storefront -> Show Swatches in Product List).
1. The product attribute uses for configurable products should be visible on Product List (Stores -> Attributes -> Product -> select attribute -> Storefront Properties -> Used in Product Listing) in order to show attribute swatches for Product List.
1. Create few configurable products for the attribute with 'swatch' (Visual Swatch, Text Swatch) type with associated products:
   - Configurable A:
      - Black A: price 1, MAP 11. (ex: 'black' color)
      - White A: price 2, MAP 12. (ex: 'white' color)
   - Configurable B:
      - Black B: price 10, MAP 110. (ex: 'black' color)
      - White B: price 20, MAP 120. (ex: 'white' color)
1. Create category, assign created products.
![Expected result](https://user-images.githubusercontent.com/1982263/71999059-695e2380-3249-11ea-9139-ca980c5f6e10.png)

### Questions or comments
Better to test once the changes from the #26240 will be merged.

### Contribution checklist (*)
